p2panda-rs/src/document/document_view_id.rs

```diff
@@ -136,4 +166,46 @@ mod tests {
             assert!(hash.validate().is_ok());
         }
     }
+
+    #[rstest]
+    fn equality(
+        #[from(random_operation_id)] operation_id_1: OperationId,
+        #[from(random_operation_id)] operation_id_2: OperationId,
+    ) {
+        let view_id_1 = DocumentViewId::new(&[operation_id_1.clone(), operation_id_2.clone()]);
+        let view_id_2 = DocumentViewId::new(&[operation_id_2, operation_id_1]);
+        assert_eq!(view_id_1, view_id_2);
+    }
+
+    #[rstest]
+    fn document_view_hash(
+        #[from(random_operation_id)] operation_id_1: OperationId,
+        #[from(random_operation_id)] operation_id_2: OperationId,
+    ) {
+        let view_id_1 = DocumentViewId::new(&[operation_id_1.clone(), operation_id_2.clone()]);
+        assert!(view_id_1.hash().validate().is_ok());
+
+        let view_id_2 = DocumentViewId::new(&[operation_id_2, operation_id_1]);
+        assert_eq!(view_id_1.hash(), view_id_2.hash());
+    }
+
+    #[test]
+    fn deserialize_unsorted_view_id() {
+        // Unsorted operation ids in document view id array:
+        //
+        // [
+        //  "0020c13cdc58dfc6f4ebd32992ff089db79980363144bdb2743693a019636fa72ec8",
+        //  "00202dce4b32cd35d61cf54634b93a526df333c5ed3d93230c2f026f8d1ecabc0cd7"
+        // ]
+        let unsorted_operation_ids = "827844303032306331336364633538646663366634656264333239393266663038396462373939383033363331343462646232373433363933613031393633366661373265633878443030323032646365346233326364333564363163663534363334623933613532366466333333633565643364393332333063326630323666386431656361626330636437";
+
+        // Construct document view id by deserialising CBOR data
+        let view_id_1: DocumentViewId =
+            ciborium::de::from_reader(&hex::decode(unsorted_operation_ids).unwrap()[..]).unwrap();
+
+        assert_eq!(
+            format!("{}", view_id_1.validate().unwrap_err()),
```

Comment:
I don't like having to use `format` to check that the correct error is returned. For our errors to implement `Eq` we need to add that to YASMF first.

Reply:
Ja, matching error types in rust is surprisingly hard ..

https://zhauniarovich.com/post/2021/2021-01-testing-errors-in-rust/

Reply:
Hmm, do you mean we shouldn't do that?

Reply:
I made a quick pr [here](https://github.com/bamboo-rs/yasmf-hash/pull/3) but I also read [this](https://users.rust-lang.org/t/help-understanding-io-error-and-lack-of-partialeq/13212) and now I'm not sure I understand the situation :D

Reply:
> Hmm, do you mean we shouldn't do that?

Ah, no, just wanted to link to a good article to show the problem space around error types in Rust. No real opinion on whats best to do, I'm fine with `format!` for now.

Reply:
Most of the time we just do `.is_err()` so checking for something more is already 100% better :joy_cat: